src/nvim/eval/typval_encode.h

```diff
@@ -154,6 +154,7 @@ typedef enum {
 /// Structure representing current VimL to messagepack conversion state
 typedef struct {
   MPConvStackValType type;  ///< Type of the stack entry.
+  typval_T *tv;  ///< Currently converted typval_T.
```

Comment:
@oni-link It is used in clear_tv only, and it is inconvenient to use in other places. Also it cannot possibly replace `data` completely, only `data.l.list` and `data.d.dict`.